src/main/java/roomescape/repository/ReservationTimeJDBCDao.java

```diff
@@ -0,0 +1,75 @@
+package roomescape.repository;
+
+import java.sql.Time;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Objects;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import roomescape.entity.ReservationTime;
+import roomescape.exceptions.EntityNotFoundException;
+
+@Repository
+public class ReservationTimeJDBCDao implements ReservationTimeRepository {
+
+    private final NamedParameterJdbcTemplate namedJdbcTemplate;
+
+    public ReservationTimeJDBCDao(NamedParameterJdbcTemplate namedJdbcTemplate) {
+        this.namedJdbcTemplate = namedJdbcTemplate;
+    }
+
+    @Override
+    public ReservationTime findById(Long id) {
+        String sql = "select * from reservation_time where id = :id";
+        MapSqlParameterSource params = new MapSqlParameterSource("id", id);
+        return namedJdbcTemplate.queryForObject(sql, params, getReservationTimeRowMapper());
+    }
+
+    @Override
+    public List<ReservationTime> findAll() {
+        String sql = "select * from reservation_time";
+        return namedJdbcTemplate.query(sql, getReservationTimeRowMapper());
+    }
+
+    @Override
+    public ReservationTime save(ReservationTime reservationTime) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        String sql = "insert into reservation_time (start_at) values (:startAt)";
+        MapSqlParameterSource params = new MapSqlParameterSource("startAt", Time.valueOf(reservationTime.startAt()));
```

Comment:
```suggestion
        SqlParameterSource params = new MapSqlParameterSource("startAt", Time.valueOf(reservationTime.startAt()));
```
java collection framework 에서 `List` 를 사용할 때처럼,
다른 api 도 구현체 대신 인터페이스로 선언해주면 좋을 듯 하네요.

Reply:
MapSqlParameterSource에 대한 고민 부족
추가로 BeanPropertySqlParameterSource 사용과 Reservation을 가상 프로퍼티 활용을 위해 일반 객체로 수정